src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -21,6 +21,10 @@ public enum ExportForm
 
     public abstract class CompilationModuleGroup
     {
+        /// <summary>
+        /// If true, "module" is in the set of input assemblies being compiled
+        /// </summary>
+        public abstract bool ContainsModule(ModuleDesc module);
```

Comment:
Is there any way to avoid adding this API to the general purpose compilation group (e.g. by adding a derived class that all R2R compilation groups need to derive from, or an interface)? Compilation group tries to avoid APIs dealing with modules on purpose: we want to support compilation modes where we only compile select types from a module within a single compilation and compile other types in a different compilation. We considered such mode for Project N in the past.

The AOT compilation group currently supports such modes, but adding APIs that deal with modules would lead to problems, should the AOT parts start depending on the new API. APIs that deal with modules are pretty hard to define in such mode (because the answer is not a `bool`, but `true`/`false`/`kindof`.

Not having a module based API was the forcing function for the current shape of the compilation group API surface.